tests/index.test.ts

```diff
@@ -387,3 +428,94 @@ describe.each([
     });
   }
 );
+
+interface MinProvider {
+  request(args: { method: string; params: any }): Promise<any>;
+}
+// example of constructing a custom contractCaller with provider only
+function constructProviderOnlyContractCaller(
```

Comment:
maybe would be good to put this guy into `/examples/` as a sample of custom contract caller

Reply:
I'd be wary of encouraging use of this helper func with any other provider. There are just to many variables that could differ. It works for `ganache` because it was written for `ganache`. You can't just pass `window.ethereum` to it an expect results.